src/Intl/Idn/bootstrap80.php

```diff
@@ -118,8 +118,8 @@
 }
 
 if (!function_exists('idn_to_ascii')) {
-    function idn_to_ascii(?string $domain, ?int $flags = IDNA_DEFAULT, ?int $variant = INTL_IDNA_VARIANT_UTS46, &$idna_info = null): string|false { return p\Idn::idn_to_ascii((string) $domain, (int) $flags, (int) $variant, $idna_info); }
+    function idn_to_ascii(string $domain, int $flags = IDNA_DEFAULT, int $variant = INTL_IDNA_VARIANT_UTS46, &$idna_info = null): string|false { return p\Idn::idn_to_ascii($domain, $flags, $variant, $idna_info); }
 }
 if (!function_exists('idn_to_utf8')) {
-    function idn_to_utf8(?string $domain, ?int $flags = IDNA_DEFAULT, ?int $variant = INTL_IDNA_VARIANT_UTS46, &$idna_info = null): string|false { return p\Idn::idn_to_utf8((string) $domain, (int) $flags, (int) $variant, $idna_info); }
+    function idn_to_utf8(string $domain, int $flags = IDNA_DEFAULT, int $variant = INTL_IDNA_VARIANT_UTS46, &$idna_info = null): string|false { return p\Idn::idn_to_utf8($domain, $flags, $variant, $idna_info); }
```

Comment:
I reverted the casts

Reply:
You need to revert the whole change. The most important part of my comment is about the parameter type.
PHP does not support passing `null` to a non-nullable *userland* function parameter. The reason of the PHP 8.1 deprecation is precisely because the behavior of native functions cannot be emulated exactly in userland for now (in PHP 9, they will behave the same than userland functions in that regard)

Reply:
@nicolas-grekas what do you think about adding legacy tests covering passing `null` to all our polyfilled functions (that will have a non-nullable parameter type in PHP 9), so that the testsuite catches such broken changes to the polyfills, as that's not the first time we get such changes in a PR.

Reply:
I see what you mean, but what about [this warning](https://github.com/symfony/polyfill/actions/runs/10795014872/job/29940652441#step:5:733)? This is the reason I updated the signature

Reply:
@alexandre-daubois this should have been fixed by https://github.com/symfony/polyfill/pull/495 AFAIK

Reply:
It doesn't seem to be touching anything related to the `Incompatible signature for PHP >= 8` warning. Am I missing something?

Reply:
it touches the detection logic at https://github.com/symfony/polyfill/pull/495/files#diff-05ed2f919a0a2d68912601230246f2c39077cd90c3ac9df8ac74651dd7c197aeR140